3_visualize/src/drought_learner_viz_components.R

```diff
@@ -1,31 +1,90 @@
 blank_plot <- function(streamflow_df, dv_tibble, growing_season = T){
   
+  axis_rects <- data.frame(
+    ymax = rep(-70, 6),
+    ymin = rep(-100, 6),
+    xmin = c(as.Date(sprintf("01/05/%s", focal_year),'%d/%m/%Y'),
+             as.Date(sprintf("01/06/%s", focal_year),'%d/%m/%Y'),
+             as.Date(sprintf("01/07/%s", focal_year),'%d/%m/%Y'),
+             as.Date(sprintf("01/08/%s", focal_year),'%d/%m/%Y'),
+             as.Date(sprintf("01/09/%s", focal_year),'%d/%m/%Y'),
+             as.Date(sprintf("01/10/%s", focal_year),'%d/%m/%Y')),
+    xmax = c(as.Date(sprintf("31/05/%s", focal_year),'%d/%m/%Y'),
+             as.Date(sprintf("30/06/%s", focal_year),'%d/%m/%Y'),
+             as.Date(sprintf("31/07/%s", focal_year),'%d/%m/%Y'),
+             as.Date(sprintf("31/08/%s", focal_year),'%d/%m/%Y'),
+             as.Date(sprintf("30/09/%s", focal_year),'%d/%m/%Y'),
+             as.Date(sprintf("14/10/%s", focal_year),'%d/%m/%Y'))
+  )
+  axis_rects_year <- data.frame(
+    ymax = rep(-70, 12),
+    ymin = rep(-100, 12),
+    xmin = c(as.Date(sprintf("01/01/%s", focal_year),'%d/%m/%Y'),
+             as.Date(sprintf("01/02/%s", focal_year),'%d/%m/%Y'),
+             as.Date(sprintf("01/03/%s", focal_year),'%d/%m/%Y'),
+             as.Date(sprintf("01/04/%s", focal_year),'%d/%m/%Y'),
+             as.Date(sprintf("01/05/%s", focal_year),'%d/%m/%Y'),
+             as.Date(sprintf("01/06/%s", focal_year),'%d/%m/%Y'),
+             as.Date(sprintf("01/07/%s", focal_year),'%d/%m/%Y'),
+             as.Date(sprintf("01/08/%s", focal_year),'%d/%m/%Y'),
+             as.Date(sprintf("01/09/%s", focal_year),'%d/%m/%Y'),
+             as.Date(sprintf("01/10/%s", focal_year),'%d/%m/%Y'),
+             as.Date(sprintf("01/11/%s", focal_year),'%d/%m/%Y'),
+             as.Date(sprintf("01/12/%s", focal_year),'%d/%m/%Y')),
```

Comment:
it would be more efficient and easier to modify using `seq` and `lubridate`
```
tibble(xmin = seq.Date(as.Date(sprintf('01/01/%s', focal_year),'%d/%m/%Y'), 
         as.Date(sprintf('01/12/%s', focal_year),'%d/%m/%Y'), 
         by = '1 month'),
       xmax = xmin+lubridate::period("1 month")-lubridate::period("1 day"))

Reply:
Thanks! I figured there would be a more efficient way to do this. Your suggestion worked perfectly.